repos/system_upgrade/el8toel9/actors/checkcustomnetworkscripts/libraries/customnetworkscripts.py

```diff
@@ -0,0 +1,52 @@
+import os
+
+from leapp import reporting
+
+
+def generate_report():
+    """ Generate reports informing user about possible manual intervention requirements """
+
+    DOC_URL = "https://red.ht/upgrading-RHEL-8-to-RHEL-9-network-scripts"
+
+    # Show documentation url if custom network-scripts detected
+    title = "custom network-scripts detected"
+    summary = "RHEL 9 does not support the legacy network-scripts package that was \
+deprecated in RHEL 8, and the custom network-scripts have been detected. In order to \
+proceed with the upgrade, you can move the custom network-scripts, for example, to \
+the `/opt/network-scripts/` directory."
+
+    reporting.create_report(
+        [
+            reporting.Title(title),
+            reporting.Summary(summary),
+            reporting.Remediation(
+                hint=(
+                    "Migrate the custom network-scripts to NetworkManager dispatcher"
+                    " scripts manually before the ugprade. Follow instructions in the"
+                    " official documentation."
+                )
+            ),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Flags([reporting.Flags.INHIBITOR]),
+            reporting.Tags([reporting.Tags.SERVICES]),
+            reporting.ExternalLink(
+                title="Upgrading from RHEL 8 to 9 - migrating custom network-scripts \
+to NetworkManager dispatcher scripts",
+                url=DOC_URL,
+            ),
+            reporting.RelatedResource("file", "/sbin/ifup-local"),
+            reporting.RelatedResource("file", "/sbin/ifup-pre-local"),
+            reporting.RelatedResource("file", "/sbin/ifdown-local"),
+            reporting.RelatedResource("file", "/sbin/ifdown-pre-local"),
+        ]
+    )
+
+
+def process():
+    if (
+        os.path.isfile("/sbin/ifup-local")
+        or os.path.isfile("/sbin/ifup-pre-local")
+        or os.path.isfile("/sbin/ifdown-local")
+        or os.path.isfile("/sbin/ifdown-pre-local")
```

Comment:
Ideally, the list of filenames should be defined only once, for example:

```python
CUSTOM_NETWORK_SCRIPTS = ["/sbin/ifup-local", ...]
...
for file in CUSTOM_NETWORK_SCRIPTS:
    reporting.RelatedResource("file", file)
...
if any(os.path.isfile(file) for file in CUSTOM_NETWORK_SCRIPTS):
    generate_report()
```

Reply:
I updated the patch, thanks.